Api schema generation

elmcrest opened the pull request:
tests passed

devkabiir committed:
🐛fix (equipment) Add correct response schema for `refuelingstops`

This tells swagger generator that the response from `refuelingstops` is an array of `RefuelingStop`

devkabiir committed:
🐛fix (company) Include `name` field in serialized data

#time: 16 minutes

#terminal_time: 2 hours 20 minutes

devkabiir committed:
🐛fix (user) Add new `GET` operation for token with correct schema

#time: 2 hours 23 minutes

#terminal_time: 2 hours 25 minutes

devkabiir committed:
🐛fix (user) Add request/reponse schema for `POST` operation

#time: 3 hours 20 minutes

#terminal_time: 3 hours 8 minutes

devkabiir committed:
🐛fix (user) Serialize query parameters instead of body in `GET` operation

#time: 3 hours 6 minutes

#terminal_time: 3 hours 11 minutes

devkabiir committed:
✅test (user) Add `test_user_gets_authenticated_using_get_operation`

#time: 14 minutes

#terminal_time: 3 hours 21 minutes

devkabiir committed:
📝docs (swagger.json) Start tracking swagger schema changes

#time: 1 hour 15 minutes

#terminal_time: 3 hours 26 minutes

elmcrest committed:
Merge pull request #370 from devkabiir/refuelingstops_api_fix

🐛fix (equipment) Add correct response schema for `refuelingstops`

devkabiir committed:
🐛fix (user) Don't generate additional tables for intermediate request/response models

#time: 32 minutes

#terminal_time: 1 hour 1 minute

devkabiir committed:
🛠chore (swagger.json) Re-generate

#time: 19 minutes

#terminal_time: 4 minutes

devkabiir commented on systori/apps/equipment/api.py:
```diff
@@ -15,7 +17,11 @@ class EquipmentModelViewSet(viewsets.ModelViewSet):
 
+    @swagger_auto_schema(responses={200: refuelingstops_response})
     @action(detail=True)
     def refuelingstops(self, request, pk=None):
```
Currently, this method returns all of the `RefuelingStop`s, as a result the output json contains an array of `RefuelingStop`s. Is there a way to limit or paginate this response? Like how most of the `list` operations work. Since this is in fact a `list` operation.

devkabiir committed:
♻️refactor (equipment) Include `equipment_type`, `maintenance`, `icon` endpoints

Also added:
1. Tests for equipment icon update/get
2.`SwaggerSchema` view to apply/add custom
modifications to enerated swagger.json

#time: 9 hours 16 minutes

#terminal_time: 7 hours 32 minutes

devkabiir committed:
✨feat (equipment) Add `DELETE` operation for icon endpoint

Also remove the unnecessary try-catch block in the `test_can_set_icon`

#time: 38 minutes

#terminal_time: 1 hour 17 minutes

elmcrest committed:
Merge pull request #371 from devkabiir/token_api_fix

Token api fix

elmcrest committed:
Merge pull request #373 from devkabiir/equipment_api_update

♻️refactor (equipment) Include `equipment_type`, `maintenance`, `icon` endpoints
